### PR TITLE
Supplemental Claims | Add evidence from private records

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

This work adds 4 new pages to the Supplemental Claims flow.
- Yes/no question asking if the Veteran wants to include evidence from a private facility or provider
- HIPAA acknowledgement page with checkbox
- A list loop page using a new pattern that allows building an array from entries that ask for the private facility/provider information including name, address, conditions treated and date range of treatment
- A final page asking if there are any limitations on the evidence that the private facility/provider will share

These pages will provide data that will be used to fill out a form 4142/4142a by our backend.

## Related issue(s)

[#49539](https://github.com/department-of-veterans-affairs/va.gov-team/issues/49539)

## Testing done

- *Describe what the old behavior was prior to the change*
  The previous page was using a [traditional list loop](https://depo-platform-documentation.scrollhelp.site/developer-docs/va-forms-library-how-to-work-with-array-data-aka-l) method; but to follow the more accessible pattern of asking one question (or group of questions) per page, we switched to using the internal [sub-task pattern](https://depo-platform-documentation.scrollhelp.site/developer-docs/va-forms-library-sub-tasks) (vs sub-task shown prior to starting a form)
- *Describe the steps required to verify your changes are working as expected*
  - Pull this branch locally, or test in the review instance
  - Log into one of the [decision review mock users](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-HLR.md)
  - Go through the flow until you get to the `/supporting-evidence/request-private-medical-records` page
  - Choose "Yes"
  - "Continue" to the next page
  - On the HIPAA acknowledgement page, check the checkbox at the end of all the content
  - "Continue" to the next page
  - Add provider info, only required fields are needed
  - Use the "Add another location" action link to add a new entry
  - Add more (unique) provider info
  - Test "Back" to go back to the previous entry
  - Use "Continue" to return to the last page
  - Use the "Add another location" action link again
  - Now use "Continue" to see a modal pop up asking if you want to keep or discard the empty entry
  - Choose "No, remove this location"
  - You should now be on the limit the information page
  - Optionally enter content
  - Done
- *Describe the tests completed and the results*
  - Added unit tests for all new pages
  - Updated e2e test to process new pages
  - All tests passing (e2e test fails to submit because of additional work pending)

## Screenshots
_Note: This field is mandatory for component work and UI changes (non-component work should NOT have screenshots)._

### Before

<details><summary>Single page list loop pattern</summary>

supporting evidence page with 2 entries. Both cards are expanded asking for name, start and end treatement dates and facility address. An add another provider facility button is under these and when used will add a third in page card. Below the button is a checkbox asking if the Veteran wants to limit their consent. When checked, it expands to reveal an input. Below that is an additional info expanded to describe what the limitation does. The overall page spans multiple screens vertically</details>

### After

<details><summary>Request private evidence</summary>

A yes-no radio pair asking if the Veteran wants to include evidence from a private provider or facility</details>

<details><summary>HIPAA acknowledgement</summary>

Lots of text associated with a checkbox asking the Veteran to acknowledge the HIPAA privacy statement</details>

<details><summary>Add facility or provider</summary>

Add facility or provider page asking for the name, address, conditions that were treated, start and end treatment dates. Add add another location action link opens an empty form page with the same elements. A back and continue button allow navigating between pages and entries</details>

<details><summary>Modal shown when navigating from an incomplete add facility or provider page</summary>

Modal asking if the Veteran wants to keep the current entry. If yes, the form will navigate back, but not navigate if going forward or adding a new entry. If no, the entry is removed and the navigation occurs</details>

<details><summary>Evidence limitation</summary>

A page with an optional textarea asking the Veteran if they want to limit the evidence provided to the VA</details>

## What areas of the site does it impact?

These pages are using a new one-question/topic per page pattern. It will be a useful replacement for other forms that use the in-page list loop pattern. This work was an initial attempt and would benefit from code refinement in both the app and in the platform code.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

- The content added in this PR may be out of sync with the latest [content source of truth](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/decision-reviews/Supplemental-Claims/design/sourceofcontenttruth.md#step-3-of-4-new-and-relevant-evidence-6) as there are some planned revisions
- [Design](https://www.sketch.com/s/d2416db4-9a4f-4919-abe4-20ba4bdcfd89/p/274ABD98-987B-4C4D-BB93-8FCE3DD6C3DE/canvas#Version)
- The evidence upload and summary pages are still a work-in-progress.

### CODE REVIEW

**Review by rocio-desantiago — APPROVED** [review]

**Review by adamwhitlock1 — APPROVED** [review]
Its interesting that authd exp FE is a blocking reviewer on this. Are we assigned as a code owner for the appeals app? I didnt see any code that touched the authd exp codebase at all.

**Comment by Mottie** [comment]
@adamwhitlock1 Oh, I don't think we needed your review to unblock this PR, but thank you!

**Pull request merged**
